spec/src/main/asciidoc/servlet-spec-body.adoc

```diff
@@ -1303,6 +1303,169 @@ the header value to an `int`, a `NumberFormatException` is thrown. If
+
+. **Mapping URI to context and resource.**
++
+The decoded path is used to map the request to a context and resource within the context. This form of the URI path is used for all subsequent mapping (web applications, servlet, filters and security constraints).
+
+. **Rejecting Suspicious Sequences.**
++
+If suspicious sequences are discovered during the prior processing steps, the request must be rejected with a 400 bad request rather than dispatched to the target servlet. If a context is matched then the error handling of the context may be used to generate the 400 response. By default, the set of suspicious sequences is defined below, but may be configured differently by a container:
+
+  * Any path not starting with the `"/"` character (e.g. `path/info`)
+  * Any path starting with an initial segment of `".."` (e.g. `/../path/info`)
+  * The encoded `"/"` character (e.g. `/path%2Finfo`)
+  * Any `"."` or `".."` segment that had a path parameter (e.g. `/path/..;/info`)
+  * Any `"."` or `".."` segment with any encoded characters (e.g. `/path/%2e%2e/info`)
+  * Any `".."` segment preceded by an empty segment (e.g. `/path//../info`)
+  * Any empty segment with parameters (e.g. `/path/;param/info` )
+  * The `"\"` character encoded or not. (e.g. `/path\info`)
+  * Any control characters either encoded or not.  (e.g. `/path%00/info`)
+  * Any illegal hex sequences following a % character
+  * Any illegal UTF-8 code sequences.
+
+==== Example URIs
+> TODO more work needed on this section. Perhaps some parts need to be moved to other sections.
+
+Consider the following URIs with a forbidden security constraint at `/secret/*` and a servlet registered at /dispatch/* that forwards request to the string return from pathInfo()
+
+. Example URIs
+|===
+| Encoded URI path  | Decoded Path | Rejected
+
+| `foo/bar` | `/foo/bar` | 400 must start with /
+| `/foo/bar` | `/foo/bar` |
+| `/foo/bar/` | `/foo/bar/` |
+| `/foo;/bar;` | `/foo/bar` |
+| `/foo;/bar;/;` | `/foo/bar` | 400 empty segment with parameters
+| `/foo%00/bar/` | `/foo/bar/` | 400 control character
+| `/foo%7F/bar/` | `/foo/bar/` | 400 control character
+| `/foo%2Fbar` | `/foo/bar` | 400 encoded /
+| `/foo\bar` | `/foo\bar` | 400 backslash character
+| `/foo%5Cbar` | `/foo\bar` | 400 backslash character
+| `/foo;%2E/bar` | `/foo/bar` |
+| `/foo;%2F/bar` | `/foo/bar` | 400 encoded /
+| `/foo/./bar` | `/foo/bar` |
+| `/foo/././bar` | `/foo/bar` |
+| `/./foo/bar` | `/foo/bar` |
+| `/foo/%2e/bar` | `/foo/bar` | 400 encoded dot segment
+| `/foo/.;/bar` | `/foo/bar` | 400 dot segment with parameter
+| `/foo/%2e;/bar` | `/foo/bar` | 400 encoded dot segment
+| `/foo/.%2Fbar` | `/foo/./bar` | 400 encoded /
+| `/foo/.%5Cbar` | `/foo/.\bar` | 400 backslash character
+| `/foo/bar/.` | `/foo/bar` |
+| `/foo/bar/./` | `/foo/bar/` |
+| `/foo/bar/.;` | `/foo/bar` | 400 dot segment with parameter
+| `/foo/bar/./;` | `/foo/bar` | 400 empty segment with parameters
```

Comment:
This (and a couple of variations) are the only query I have at the moment. I think the decoded path should end with `"/"`.
(see above)

Reply:
+1
As should the ``/foo;/bar;/;`` above, which should also not be reject with with empty segment.   
I'm going to try to fix with last empty segment handling.... stand by....

Reply:
I have pushed an alternative that treats any trailing `/` as an empty segment.  I think it is clearer and simpler?     Unit test now passes all the asserts